src/lib/types/model.ts

```diff
@@ -124,12 +124,10 @@ export interface IEnvironment {
     protected: boolean;
 }
 
-export interface IEnvironmentCreate {
-    name: string;
-    type: string;
-    sortOrder?: number;
-    enabled?: boolean;
-}
+/*
+ * @deprecated prefer EnvironmentCreateSchema type instead
+ */
+export interface IEnvironmentCreate extends EnvironmentCreateSchema {}
```

Comment:
This is just a way of maintaining backward compatibility. There are other options:
1. To just replace `IEnvironmentCreate` with `EnvironmentCreateSchema` (fixing the code: imports and type references)
2. To remove `IEnvironmentCreate` and rename `EnvironmentCreateSchema` to `IEnvironmentCreate` fixing the imports
3. Keep `IEnvironmentCreate` interface and use [typescript to json schema](https://github.com/YousefED/typescript-json-schema#programmatic-use) removing the need of `EnvironmentCreateSchema` type.

Reply:
> 1. To just replace IEnvironmentCreate with EnvironmentCreateSchema (fixing the code: imports and type references)

I'd go with this one: remove the old type and use the new one instead. Should be a pretty safe refactor.

Reply:
Thanks @olav wdyt about the naming? Are we ending the type name with `Schema` as a naming convention?

Reply:
I like type aliases for cases like this, personally, so I'm for it 😄